### PR TITLE
Add postgres as a db

### PR DESCRIPTION
**Pull request by evkuzin**
`github.com/lib/pq` is in maintenance mode, but it seems like the best candidate still because we don't need all these fancy features that pgx provides and its fits as a database/sql Interface. Basically, I chose it based on [this](https://github.com/jackc/pgx#choosing-between-the-pgx-and-databasesql-interfaces)
+ @juliusv

### CODE REVIEW

File: pkg/sharer/sharer.go [hunk 1]
```diff
@@ -238,13 +266,15 @@ func NewSQLSharer(logger log.Logger, driver string, dsn string, createTables boo
 }
 
 func (s SQLSharer) cleanupOldLinks(retention time.Duration) (int64, error) {
-	timestampFn := "DATETIME"
-	if s.driver == "mysql" {
-		timestampFn = "TIMESTAMP"
+	var query string
+	if s.driver == "postgres" {
```
Comment (juliusv) [inline pkg/sharer/sharer.go:270]: With three variants, a `switch` starts to feel more natural:

```go
switch s.driver {
case "postgres":
  // ...
case "mysql":
  // ...
default:
  // ...
}
```

File: .gitignore [hunk 2]
```diff
@@ -1,2 +1,3 @@
 /promlens
 /npm_licenses.tar.bz2
+.idea
```
Comment (juliusv) [inline .gitignore:3]: Hmm, let's either remove this unrelated change from the PR or end the file with a new line at least.

Comment (evkuzin) [reply .gitignore]: If you don't mind I'd love to keep it. I like the project and would be glad to contribute more, this is just an IDE settings folder.

File: pkg/sharer/sharer.go (hunk 1 above)
Comment (evkuzin) [reply pkg/sharer/sharer.go]: Right, thank you for pointing. Fixed

File: .gitignore (hunk 2 above)
Comment (Nexucis) [reply .gitignore]: yeah but Prometheus never ignore IDE settings folder like IntelliJ or Eclipse.
Instead you will have to use a global `.gitignore` file, like that: https://stackoverflow.com/questions/7335420/global-git-ignore

For example on my local environment:

```bash
cat .gitconfig
[pull]
	ff = only
[user]
	name = Augustin Husson
	email = husson.augustin@gmail.com
[credential]
	helper = store --file ~/.git/credential
[pager]
	branch = false
[core]
	excludesFile = ~/.git/.gitignore
```

``` bash
cat .git/.gitignore
.idea/
```

Comment (evkuzin) [reply .gitignore]: TIL about global .gitignore file. Thanks! updated.

**Comment by juliusv** [comment]
:+1: Wonderful, thank you :)
